src/main/java/teammates/ui/controller/InstructorEditInstructorFeedbackSaveAction.java

```diff
@@ -0,0 +1,195 @@
+package teammates.ui.controller;
+
+import teammates.common.datatransfer.FeedbackParticipantType;
+import teammates.common.datatransfer.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.datatransfer.FeedbackSessionQuestionsBundle;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.common.util.HttpRequestHelper;
+import teammates.common.util.StatusMessage;
+import teammates.common.util.Const.StatusMessageColor;
+import teammates.logic.api.GateKeeper;
+
+/**
+ * The {@code InstructorEditInstructorFeedbackSaveAction} class handles incoming requests to 
+ * save the data after moderating the instructor.
+ */
+public class InstructorEditInstructorFeedbackSaveAction extends FeedbackSubmissionEditSaveAction {
+    InstructorAttributes moderatedInstructor;
+    
+    /**
+     * Verifies if the user is allowed to carry out the action.
+     */
+    @Override
+    protected void verifyAccesibleForSpecificUser() {
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
+        FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+
+        new GateKeeper().verifyAccessible(instructor,
+                session,
+                false, 
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS);
```

Comment:
Hmm. I was looking at the permission when instructors are moderating students and it uses this for saving of the moderated values and loading of the feedback form. Should I add new permission for instructor to moderate students and instructor to moderate instructor because this also doesn't seem like the permission should used when instructor moderates student. Should all these be done in a separate PR? Or I just include them here? @LowWeiLin